src/app/ui/admin-page/add-badge-dialog/add-badge-dialog.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
+
+@Component({
+  selector: 'add-badge-dialog',
+  templateUrl: './add-badge-dialog.component.html',
+  styleUrls: ['./add-badge-dialog.component.scss']
+})
+export class AddBadgeDialogComponent implements OnInit {
+  badgeForm: FormGroup;
+  constructor(private dialogRef: MatDialogRef<AddBadgeDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data) { }
```

Comment:
We need only uid as data input right? Also you can define the dialog ref output via:
```suggestion
  constructor(
    private dialogRef: MatDialogRef<AddBadgeDialogComponent, Badge>,
    @Inject(MAT_DIALOG_DATA) public uid: string
  ) { }
```